chains/evm/config.go

```diff
@@ -0,0 +1,124 @@
+package evm
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ChainSafe/chainbridge-core/chains"
+)
+
+const DefaultGasLimit = 6721975
+const DefaultGasPrice = 20000000000
+const DefaultGasMultiplier = 1
+
+// Chain specific options
+var (
+	BridgeOpt         = "bridge"
+	Erc20HandlerOpt   = "erc20Handler"
+	Erc721HandlerOpt  = "erc721Handler"
+	GenericHandlerOpt = "genericHandler"
+	MaxGasPriceOpt    = "maxGasPrice"
+	GasLimitOpt       = "gasLimit"
+	GasMultiplier     = "gasMultiplier"
+	HttpOpt           = "http"
+	// StartBlockOpt         = "startBlock"
+	// BlockConfirmationsOpt = "blockConfirmations"
+	// EGSApiKey             = "egsApiKey"
+	// EGSSpeed              = "egsSpeed"
+)
+
+type EVMConfig struct {
+	GeneralChainConfig chains.GeneralChainConfig
+	Bridge             string
+	Erc20Handler       string
+	Erc721Handler      string
+	GenericHandler     string
+	MaxGasPrice        *big.Int
+	GasMultiplier      *big.Float
+	GasLimit           *big.Int
+	Http               bool
+}
+
+func ParseConfig(rawConfig *chains.RawChainConfig) (*EVMConfig, error) {
+	config := &EVMConfig{
+		GeneralChainConfig: rawConfig.GeneralChainConfig,
+		Bridge:             "",
+		Erc20Handler:       "",
+		Erc721Handler:      "",
+		GenericHandler:     "",
+		GasLimit:           big.NewInt(DefaultGasLimit),
+		MaxGasPrice:        big.NewInt(DefaultGasPrice),
+		GasMultiplier:      big.NewFloat(DefaultGasMultiplier),
+		Http:               false,
+	}
+
+	if contract, ok := rawConfig.Opts[BridgeOpt]; ok && contract != "" {
```

Comment:
I wanted to have a shared general configuration struct by both substrate and ethereum, including the options field. When starting the parser it seemed the simplest way to do this was keeping a map[string]string then parsing it. I also looked into having a separate raw evm and substrate config which included the respective opts in the struct. This could potentially make the parser methods a lot cleaner.